### PR TITLE
LA-123 Export multiple documents in MySpace iOS

### PR DESCRIPTION
**Pull request by manhduc-nguyen**

### CODE REVIEW

File: lib/presentation/widget/myspace/my_space_widget.dart [hunk 1]
```diff
@@ -509,4 +510,16 @@ class _MySpaceWidgetState extends State<MySpaceWidget> {
             itemSelectionType: ItemSelectionType.multiple))
         .build();
   }
+
+  Widget _exportMultiFilesAction(BuildContext context, List<Document> documents) {
+    return DocumentContextMenuTileBuilder(
```
Comment (hoangdat) [inline lib/presentation/widget/myspace/my_space_widget.dart:515]: IMO, you need to confirm `documents` is not empty

**Comment by manhduc-nguyen** [comment]
@hoangdat @imGok please review this guys

**Review by hoangdat — APPROVED** [review]

File: domain/lib/src/usecases/download_file/download_multiple_file_ios_interactor.dart [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  Future<Either<Failure, Success>> execute({@required List<Document> documents, @required CancelToken cancelToken}) async {
+    final listResult = await Future.wait(documents.map((element) =>
+        _downloadFileIOSInteractor.execute(element, cancelToken)));
+    if (listResult.length == 1) {
+      return listResult.first;
+    } else {
+      var failedFileCount = 0;
+      listResult.forEach((element) {
+        if (element is Left) {
```
Comment (imGok) [inline domain/lib/src/usecases/download_file/download_multiple_file_ios_interactor.dart:51]: Forgot to use it my PR but there is 
```
listResult.whereType<Left>().length 
```
which I find sexier ^^

**Review by imGok — APPROVED** [review]

File: domain/lib/src/usecases/download_file/download_multiple_file_ios_interactor.dart (hunk 2 above)
Comment (hoangdat) [reply domain/lib/src/usecases/download_file/download_multiple_file_ios_interactor.dart:51]: cool, please insert it on the fixup PR :)

**Pull request merged**